pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -47,6 +76,234 @@ def get_average_of_unique_labels(labels, values):
+
+    The interstitial sites can be obtained through `get_interstitials`
+
+    In complex structures (i.e. grain boundary, dislocation etc.), the default parameters
+    should be chosen properly. In order to see other quantities, which potentially
+    characterize interstitial sites, see the following functions from
+    `structure.analyse.interstitials`:
+
+        - `get_variance()`
+        - `get_distance()`
+        - `get_steinhardt_parameter()`
+        - `get_volume()`
+        - `get_area()`
+    """
+    def __init__(self, structure, n_gridpoints_per_angstrom=5, min_distance=1, use_voronoi=False):
+        """
+        Args:
+            structure (pyiron_atomistics.atomistics.structure.atoms.Atoms): Reference structure
+            n_gridpoints_per_angstrom (int): Number of grid points per angstrom for the
+                initialization of the interstitial candidates. The finer the mesh (i.e. the larger
+                the value), the likelier it is to find the correct sites but then also it becomes
+                computationally more expensive. Ignored if `use_voronoi` is set to `True`
+            min_distance (float): Minimum distance from the nearest neighboring atoms to the
+                positions for them to be considered as interstitial site candidates.
+            use_voronoi (bool): Use Voronoi vertices for the initial interstitial candidate
+                positions instead of grid points.
+        """
+        self.min_distance = min_distance
+        self.structure = structure
+        if use_voronoi:
+            self.positions = structure.analyse.get_voronoi_vertices()
+        else:
+            self.positions = self._create_gridpoints(
+                n_gridpoints_per_angstrom=n_gridpoints_per_angstrom
+            )
+        if min_distance > 0:
+            self._remove_too_close()
+
+    @property
+    def num_neighbors(self):
+        return self._num_neighbors
+
+    @num_neighbors.setter
+    def num_neighbors(self, n):
+        self.reset()
+        self._num_neighbors = n
+
+    def reset(self):
+        self._hull = None
+        self._neigh = None
+
+    @property
+    def neigh(self):
+        if self._neigh is None:
+            self._neigh = self.structure.get_neighborhood(
+                self.positions, num_neighbors=self.num_neighbors
+            )
+        return self._neigh
+
+    @property
+    def positions(self):
+        return self._positions
+
+    @positions.setter
+    def positions(self, x):
+        self.reset()
+        self._positions = x
+
+    @property
+    def hull(self):
+        if self._hull is None:
+            self._hull = [ConvexHull(v) for v in self.neigh.vecs]
+        return self._hull
+
+    def _create_gridpoints(self, n_gridpoints_per_angstrom=5):
+        cell = self.structure.cell.diagonal()
+        gridpoints = (n_gridpoints_per_angstrom*cell).astype(int)
+        positions = [np.linspace(0, cell[i], gridpoints[i], endpoint=False) for i in range(3)]
```

Comment:
Would that break for non-cubic cells?  What about

```python
i = np.arange(grid_points)
positions = np.einsum('ij,j... -> i...', self.structure.cell.array, np.meshgrid(i, i, i))
```

(I hope this is how einsum works)

Reply:
uh that's a super good point